tests/test_file_event_actions.py

```diff
@@ -0,0 +1,615 @@
+        self, mocker, mock_py42_client
+    ):
+        param = {"hash": TEST_SHA256, "file_name": TEST_FILENAME}
+        connector = _create_hunt_file_connector(mocker, mock_py42_client)
+        connector.handle_action(param)
+        mock_py42_client.securitydata.stream_file_by_sha256.assert_called_once_with(
+            TEST_SHA256
+        )
+        assert_success(connector)
+
+    def test_handle_action_when_hunt_file_given_sha256_and_filename_calls_create_attachment_with_expected_args(
+        self, mocker, mock_py42_client, mock_create_attachment
+    ):
+        param = {"hash": TEST_SHA256, "file_name": TEST_FILENAME}
+        connector = _create_hunt_file_connector(mocker, mock_py42_client)
+        connector.handle_action(param)
+        mock_create_attachment.assert_called_once_with(
+            b"filecontent", 42, file_name=TEST_FILENAME
+        )
+        assert_success(connector)
+
+    def test_handle_action_when_hunt_file_given_sha256_and_no_filename_calls_create_attachment_with_expected_args(
+        self, mocker, mock_py42_client, mock_create_attachment
+    ):
+        param = {"hash": TEST_SHA256}
+        connector = _create_hunt_file_connector(mocker, mock_py42_client)
+        connector.handle_action(param)
+        mock_create_attachment.assert_called_once_with(
+            b"filecontent", 42, file_name=TEST_SHA256
+        )
+        assert_success(connector)
+
+    def test_handle_action_when_hunt_file_given_sha256_and_filename_sets_success_message(
+        self, mocker, mock_py42_client
+    ):
+        param = {"hash": TEST_SHA256, "file_name": TEST_FILENAME}
+        connector = _create_hunt_file_connector(mocker, mock_py42_client)
+        connector.handle_action(param)
+        message = (
+            f"{TEST_FILENAME} was successfully downloaded and attached to container 42"
+        )
+        assert_successful_message(connector, message)
+
+    def test_handle_action_when_hunt_file_given_sha256_and_no_filename_sets_success_message(
+        self, mocker, mock_py42_client
+    ):
+        param = {"hash": TEST_SHA256}
+        connector = _create_hunt_file_connector(mocker, mock_py42_client)
+        connector.handle_action(param)
+        message = (
+            f"{TEST_SHA256} was successfully downloaded and attached to container 42"
+        )
+        assert_successful_message(connector, message)
+
+    def test_handle_action_when_hunt_file_given_unsupported_hash_sets_error_message(
+        self, mocker, mock_py42_client
+    ):
+        param = {"hash": "not-a-valid-hash"}
+        connector = _create_hunt_file_connector(mocker, mock_py42_client)
+        connector.handle_action(param)
+        message = "Code42: Failed execution of action hunt_file: Unsupported hash format. Hash must be either md5 or sha256"
+        assert_fail_message(connector, message)
+
+    def test_handle_action_when_run_query_calls_search_file_events_with_expected_query(
```

Comment:
checks to make sure each value that is passed in actually shows up in the query.